tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -149,8 +149,8 @@ void EditorDiagConsumer::handleDiagnostic(SourceManager &SM,
 
     SKInfo.Offset = SM.getLocOffsetInBuffer(Info.Loc, BufferID);
     std::tie(SKInfo.Line, SKInfo.Column) =
-        SM.getPresumedLineAndColumnForLoc(Info.Loc, BufferID);
-    SKInfo.Filename = SM.getDisplayNameForLoc(Info.Loc).str();
+        SM.getLineAndColumnInBuffer(Info.Loc, BufferID);
+    SKInfo.Filename = SM.getIdentifierForBuffer(BufferID).str();
```

Comment:
Diagnostics should use the presumed location.  The editor only shows the current file, but other things like the compile tracker notifications (which are used in the index log) will include diagnostics for all files.

I think the right fix here is that the code below that assigns to `BufferDiagnostics` should be changed to use the presumed file instead.

Reply:
Wouldn't that mean we never show those live issues though? Also if offset matters we'd need to grab it from line/col in this case.

Reply:
I see your point.  On the other hand, it's unlikely that you are _editing_ a file with source location directives at all - usually that is a generated file.  And if you get diagnostics from another file in your module that is generated, you probably want them to point to the same place as the compiler would put them.  But maybe there are cases where you're inspecting the generated file and would want to see the diagnostics - maybe we could duplicate them? Or use the buffer location for the editor diagnostic consumer but the presumed location for `getAllDiagnostics()`?

Reply:
I've just changed it back for now. It isn't really causing any bugs like the others were.